# SPDX-FileCopyrightText: Huawei Inc.
#
# SPDX-License-Identifier: Apache-2.0

Patch for //base/hiviewdfx/faultloggerd of OpenHarmony 3.0 codebase

This adds sd_notify(3) ready notification to faultlogger service for better
integration with systemd.

Signed-off-by: Esben Haabendal <esben@geanix.com>
Upstream-Status: Pending

diff --git a/services/fault_logger_daemon.cpp b/services/fault_logger_daemon.cpp
index ed7794cfbfc7..e0b35c44a48c 100644
--- a/services/fault_logger_daemon.cpp
+++ b/services/fault_logger_daemon.cpp
@@ -32,6 +32,8 @@
 #include <hilog/log.h>
 #include <securec.h>
 
+#include <systemd/sd-daemon.h>
+
 #include "fault_logger_daemon.h"
 #include "faultloggerd_client.h"
 
@@ -44,11 +46,7 @@ constexpr int32_t MAX_CONNECTION = 4;
 constexpr int32_t REQUEST_BUF_SIZE = 1024;
 constexpr int32_t MSG_BUF_SIZE = 256;
 constexpr int32_t SYSTEM_UID = 1000;
-#if defined(USE_MUSL)
-static const char FAULTLOGGERD_SOCK_PATH[] = "/dev/unix/socket/faultloggerd.server";
-#else
-static const char FAULTLOGGERD_SOCK_PATH[] = "/dev/socket/faultloggerd.server";
-#endif
+static const char FAULTLOGGERD_SOCK_PATH[] = "/run/openharmony/faultlogger/server";
 static const char FAULTLOGGERD_BASE_PATH[] = "/data/log/faultlog/temp/";
 
 static std::string GetRequestTypeName(int32_t type)
@@ -254,6 +252,7 @@ int32_t StartServer(int argc, char *argv[])
         return -1;
     }
 
+    sd_notify(0, "READY=1");
     struct sockaddr_un clientAddr;
     socklen_t clientAddrSize = sizeof(clientAddr);
     int connectionFd = -1;
diff --git a/services/BUILD.gn b/services/BUILD.gn
index f48777a7cdbf..847e3cf79510 100644
--- a/services/BUILD.gn
+++ b/services/BUILD.gn
@@ -63,6 +63,7 @@ ohos_executable("faultloggerd") {
     "$faultloggerd_path/interfaces/innerkits/faultloggerd_client:libfaultloggerd",
     "//utils/native/base:utils",
   ]
+  libs = [ "systemd" ]
 
   if (is_standard_system) {
     external_deps = [ "hiviewdfx_hilog_native:libhilog" ]
